core/src/layers/prometheus.rs

```diff
@@ -93,32 +89,117 @@ use crate::*;
 ///     println!("## Prometheus Metrics");
 ///     println!("{}", String::from_utf8(buffer.clone()).unwrap());
 ///     Ok(())
-/// }
+/// # }
 /// ```
 #[derive(Clone, Debug)]
 pub struct PrometheusLayer {
-    registry: Registry,
+    interceptor: PrometheusInterceptor,
+}
+
+impl Default for PrometheusLayer {
+    fn default() -> Self {
+        let register = prometheus::default_registry();
+        Self::new(register)
+    }
+}
+
+impl PrometheusLayer {
+    /// Create a [`PrometheusLayer`] and register its metrics to the given registry.
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// # use log::debug;
+    /// # use opendal::layers::PrometheusLayer;
+    /// # use opendal::services;
+    /// # use opendal::Operator;
+    /// # use opendal::Result;
+    /// #
+    /// # #[tokio::main]
+    /// # async fn main() -> Result<()> {
+    ///     // Pick a builder and configure it.
+    ///     let builder = services::Memory::default();
+    ///     let registry = prometheus::default_registry();
+    ///
+    ///     let op = Operator::new(builder)
+    ///         .expect("must init")
+    ///         .layer(PrometheusLayer::new(registry))
+    ///         .finish();
+    ///     debug!("operator: {op:?}");
+    ///
+    ///     Ok(())
+    /// # }
+    /// ```
+    pub fn new(registry: &Registry) -> Self {
```

Comment:
How about removing the `new` API? Offering a single method to build layers simplifies usage for users.

Reply:
You mean like this?

```rust
pub fn register_default() -> Self {
    let registry = prometheus::default_registry();
    PrometheusLayerBuilder::default().register(registry)
}
```